include/plugin/runtime-support.h

```diff
@@ -710,6 +709,95 @@ inline ::zeek::ValPtr to_val(const hilti::rt::Set<T>& s, ::zeek::TypePtr target)
     return zv;
 }
 
+
+namespace {
+template<typename, template<typename...> typename>
+struct is_instance_impl : public std::false_type {};
+
+template<template<typename...> typename U, typename... Ts>
+struct is_instance_impl<U<Ts...>, U> : public std::true_type {};
+} // namespace
+
+template<typename T, template<typename...> typename U>
+using is_instance = is_instance_impl<std::remove_cv_t<T>, U>;
+
+
+template<typename T>
+inline void set_record_field(::zeek::RecordVal* rval, ::zeek::IntrusivePtr<::zeek::RecordType>& rtype, int idx,
```

Comment:
Is `rtype` mutable due to const-correctness issues in `IntrusivePtr`/`RecordType`?

Reply:
Oversight.